ddl/ddl_api.go

```diff
@@ -3023,6 +3084,10 @@ func (d *ddl) CreateIndex(ctx sessionctx.Context, ti ast.Ident, unique bool, ind
 	}
 
 	err = d.doDDLJob(ctx, job)
+	// key exists, but if_not_exists flags is true, so we ignore this error.
+	if ErrDupKeyName.Equal(err) && ifNotExists {
```

Comment:
Do we need to report the warning here?

Reply:
done in https://github.com/pingcap/tidb/pull/10723/commits/0d609e3ad5b32a09655a369a9cbf8b3bd17ff3b6.